lib/workers/repository/updates/generate.ts

```diff
@@ -355,6 +363,35 @@ export function generateBranchConfig(
     releaseTimestamp: releaseTimestamp!,
   }; // TODO: fixme (#9666)
 
+  // Calculate the highest priority `semanticCommitType`
+  let highestPrioritySemanticCommitType = '';
+  for (const upgrade of config.upgrades) {
+    if (
+      upgrade.semanticCommits === 'enabled' &&
+      upgrade.semanticCommitType &&
+      upgrade.semanticCommitType in semanticCommitTypePriorities
+    ) {
+      const priority = semanticCommitTypePriorities[upgrade.semanticCommitType];
+      const highestPriority =
+        semanticCommitTypePriorities[highestPrioritySemanticCommitType] ?? -1;
+
+      if (priority > highestPriority) {
+        highestPrioritySemanticCommitType = upgrade.semanticCommitType;
+      }
+    }
+  }
+
+  if (highestPrioritySemanticCommitType) {
+    config.semanticCommitType = highestPrioritySemanticCommitType;
+  }
+
+  // Enable `semanticCommits` if one of the branches has it enabled
+  if (
+    config.upgrades.some((upgrade) => upgrade.semanticCommits === 'enabled')
+  ) {
+    config.semanticCommits = 'enabled';
+  }
```

Comment:
You could maybe put this logic first and then only do the other logic if this one is enabled.